astropy/cosmology/core.py

```diff
@@ -1326,7 +1329,7 @@ def angular_diameter_distance_z1z2(self, z1, z2):
         z1 = np.atleast_1d(z1)
```

Comment:
In fact, if one removes the test, it is not necessary to make `z1` and `z2` arrays (I think). Just write the second test as
```
if np.any(z1 > z2)
```